# **Students Who Slip On Maths**

## Objectives

* Import the cleaned .csv file to a pandas dataframe for exploration

* Investigate xyz

* 



## Inputs

* The clean 'student_mat_cleaned.csv'  

* Libraires containing the tools needed, pandas, numpy, matplotlib, seaborn. 

## Outputs

* Visualisations 



---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\NateM\\Desktop\\CI_DA\\Capstone_Three\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\NateM\\Desktop\\CI_DA\\Capstone_Three'

# Import block

In [4]:
# Libraries needed 
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns
import sklearn as sk
from scipy.stats import spearmanr





---

# Section 1 - Load

Load the cleaned .csv into a pandas dataframe

In [5]:
# Load the raw data
df_cleaned = pd.read_csv("Dataset/Cleaned/student_mat_cleaned.csv")

# View the dataframe
df_cleaned.head(5)

,school,age,reason,guardian,traveltime,studytime,failures,schoolsup,famsup,paid,...,freetime,goout,Dalc,Walc,health,absences,G1,G2,G3,studytime_label
0,GP,18,course,mother,2,2,0,yes,no,no,...,3,4,1,1,3,6,5,6,6,2 - 5 hours
1,GP,17,course,father,1,2,0,no,yes,no,...,3,3,1,1,3,4,5,5,6,2 - 5 hours
2,GP,15,other,mother,1,2,3,yes,no,yes,...,3,2,2,3,3,10,7,8,10,2 - 5 hours
3,GP,15,home,mother,1,3,0,no,yes,yes,...,2,2,1,1,5,2,15,14,15,5 - 10 hours
4,GP,16,home,father,1,2,0,no,yes,yes,...,3,2,1,2,5,4,6,10,10,2 - 5 hours


---

# Section 2 - Hypothesis 1

Testing the first hypothesis using Spearman Rank Correltaion, because we want to know if as the studytime goes up, the final grade increases.

In [6]:
rho, p_value = spearmanr(
    df_cleaned["studytime"],
    df_cleaned["G3"]
)

print(f"Spearman correlation: {rho:.3f}")
print(f"P-value: {p_value:.4f}")

Spearman correlation: 0.105
P-value: 0.0367


These numbers show that there is a statistical signifcance between weekly study time and a higher final grade but they also show that it's weak, showing that studying is benficial but it isn't a driving force in getting a definitive final grade improvement. Student Support Groups can help students improve their final grade (G3) by some amount through encouraging the ones that study less than 5 hours to study for longer sum durations.     

---

# Section 3 - Hypothesis 2

Likewise, H2 will also be subject to a Spearman Rank Correlation test.

In [7]:
rho, p_value = spearmanr(
    df_cleaned["failures"],
    df_cleaned["G3"]
)

print(f"Spearman correlation: {rho:.3f}")
print(f"P-value: {p_value:.4f}")

Spearman correlation: -0.361
P-value: 0.0000


The rho of -0.361 and a p-value of 0.000 shows there is a significant and negative association between the previous class failures and a lower mathematics grade overall. This is evidence in support of H2 that previous failure does impact a students final grade, and that that impact increases as they continue to fail. 

---

# Section 4 - Hypothesis 3

Once again as we're looking to see if there's a correlation as one aspect goes up or down does the other go up or down we're goin to use Spearman Rank Correlation on absences and final grade (G3).


In [8]:
rho, p_value = spearmanr(
    df_cleaned["absences"],
    df_cleaned["G3"]
)

print(f"Spearman correlation: {rho:.3f}")
print(f"P-value: {p_value:.4f}")

Spearman correlation: 0.018
P-value: 0.7254


The near 0 correlation value indicates that there's no monotinoic relatinship between a student's rate of absence and their final grade. With a high p-value compared to the alpha of 0.05 also declaring that there is no statistical significance betweem the 2 features. Within the scope of this dataset H3 is unsupported.

---

# Section 5 - Summary

The testing identi